src/txn/txn.c

```diff
@@ -1711,9 +1712,18 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     if (!prepare)
         F_CLR(txn, WT_TXN_TS_ROUND_PREPARED);
 
+    /* Enter the commit generation before timestamp validity check. */
+    if (!prepare)
```

Comment:
I'm OK with this approach.

However, personally I prefer to treat prepare transactions the same as normal transactions. Therefore, the approach to move the code that gets the checkpoint snapshot outside of checkpoint prepare call is a better approach to me.

Reply:
@quchenhao I think the scenario we are talking about in this change does not hold good for prepared transactions because there is a check [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/txn/txn.c#L1967) in the `txn_commit` that the stable timestamp cannot be concurrently increased greater than or equal to the prepared transaction's durable timestamp. So, it is safe to skip the prepared transactions to not enter into the commit generation.